### PR TITLE
#P03. As Participant, I would like to update my profile.

### PR DESCRIPTION
**Pull request by natalijabujevic0708**
## Description
The users can edit their profile details:

- First Name
- Last Name
- Email Address
- Slack Display Name
- User type
- Current Module on LMS

## Pull request type
- #P03 (Participant User Story #03) As Participant, I would like to update my profile.

## Testing

All testing was done manually and the profile was successfully edited and updated.

## Screenshots

1. Original profile page.
hackathon-app_profile_page

2.Edit Profile
hackathon-app_edit_profile

3. Updated Profile
hackathon-app_edited_profile

## Additional Information

There is one field missing on the edit profile page:

Shipping Address

- hidden by default
- is enabled only if part of a winning team

I am planning to add it in the future as currently I don't know how to implement the condition  _enabled only if part of a winning team_

## Does this introduce a breaking change

- [ ] Yes
- [x] No

### CODE REVIEW

File: accounts/templates/accounts/edit_profile.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{% extends 'base.html' %}
+
+{% block content %}
+<h1>Edit Profile</h1>
+<div class="container">
+	<form method="post" enctype="multipart/form-data">
+		{% csrf_token %}
+		<table>
```
Comment (stefdworschak) [inline accounts/templates/accounts/edit_profile.html:8]: Why did you add a table here? I don't think that's needed.

File: accounts/templates/accounts/edit_profile.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{% extends 'base.html' %}
+
+{% block content %}
+<h1>Edit Profile</h1>
+<div class="container">
```
Comment (stefdworschak) [inline accounts/templates/accounts/edit_profile.html:5]: The base template already has an element with a `container` class, so I would remove this div.

File: accounts/views.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+@login_required
+def edit_profile(request):
+    """ 
+        If the request is POST and the form is valid, save the form and redirect to profile.
+        Otherwise, display current user instance in EditProfileForm on edit_profile.html.
+    """
+    if request.method == 'POST':
+        form = EditProfileForm(request.POST, instance=request.user)
+       
+        if form.is_valid():
```
Comment (stefdworschak) [inline accounts/views.py:15]: I think you need to add another `else` here for when `form.is_valid()` is not `True`.

File: accounts/views.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+            form.save()
+          
+            return redirect('profile')
+    else:
+        form = EditProfileForm(instance=request.user)
+        contex = {
```
Comment (stefdworschak) [inline accounts/views.py:21]: I think you mispelt context here, but I would just pass the dict directly into `render`.

File: accounts/forms.py [hunk 5]
```diff
@@ -44,3 +48,20 @@ def signup(self, request, user):
...
+    """
+
+    class Meta:
+        model = CustomUser
+        fields = (
+            'email',
+            'first_name',
+            'last_name',
+            'slack_display_name',
+            'user_type',
```
Comment (stefdworschak) [inline accounts/forms.py:65]: Could you please remove this field? I don't think the user should be able to change this. Instead a change in `user_type` should come from an admin.

**Review by stefdworschak — CHANGES_REQUESTED** [review]
@natalijabujevic0708 looking very good, there are just a few minor comments from me. Overall there are also two extra things that I would like you to look at as part of this PR. 
1) Adding user_type and current_lms_module to the profile view
2) Applying some better styling to the edit profile form

**Comment by natalijabujevic0708** [comment]
@stefdworschak I made the changes, let me know if everything is ok or I should change something :)

**Review by stefdworschak — APPROVED** [review]
LGTM

@natalijabujevic0708 we merged some more changes and it seems there is a merge conflict now, could you just resolve that and let me know once it's done so I can merge the PR?

**Comment by natalijabujevic0708** [comment]
@stefdworschak done :)

**Pull request merged**